payments/toncoin.mdx

```diff
@@ -3,6 +3,144 @@ title: "Toncoin payments processing"
 sidebarTitle: "Toncoin"
 ---
 
-import { Stub } from '/snippets/stub.jsx';
+import { Aside } from "/snippets/aside.jsx";
 
-<Stub issue="204" />
+Processing Toncoin payments requires choosing between two architectural approaches: invoice-based deposits to a single address, common to all users or unique deposit addresses per user. Each approach has different security characteristics, implementation complexity, and operational requirements.
+
+## Deposit methods comparison
+
+**Invoice-based flow**
+
+```mermaid
+graph LR
+  U1[User A] -- invoice UUID --> SW[Shared Wallet]
+  U2[User B] -- invoice UUID --> SW
+  SW --> EX[Exchange Ledger]
+```
+
+**Unique deposit address flow**
+
+```mermaid
+graph LR
+  U1[User A] --> W1[Wallet A]
+  W1 --> EX[Exchange Ledger]
+  U2[User B] --> W2[Wallet B]
+  W2 --> EX
+```
+
+**Comparison table**
+
+| Criteria                     | Invoice-based deposits                                                                                | Unique deposit addresses                                                                 |
+| ---------------------------- | ----------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------- |
+| Security exposure            | One shared hot wallet; a key leak drains the full pool                                                | Each wallet isolates funds; compromise affects only the impacted user                    |
+| User input requirements      | User must include an invoice ID comment; missing or malformed comments need manual recovery workflows | User only needs the destination address                                                  |
+| Parsing and validation       | Backend parses comments on every deposit and matches to invoices                                      | No parsing; deposit attribution is address-based                                         |
+| Deployment and storage costs | Deploy and maintain a single wallet; storage rent limited to that contract                            | Deploy many wallets; storage rent and deployment gas scale with user count               |
```

Comment:
Need a link to the storage rent explanation. This is more or less unique to TON blockchain